src/pages/recipes/index.md

```diff
@@ -451,3 +451,394 @@ function main(){
+1. Document name
+2. Application name
+
+Use the following script to remove the event listeners added by the preceding script:
+
+```js
+app.removeEventListener("beforeImport", myEventInfo);
+app.documents.item(0).removeEventListener("beforeImport", myEventInfo);
+```
+
+### Writing Events Script to Run Through Out the Session
+After the event listener is associated, the script is blocked by a promise and wait for the event in the background until the event listener is removed. If the event occurs, there is an automatic callback to execute the related function. If the event listener is removed, the promise can be resolved to complete the script execution.
+
+Here is a sample events script to run and listen for “afterNew” event throughout the InDesign session and generate an alert whenever a new document is created.
+
+```js
+let myInDesign = require("indesign");
+let app = myInDesign.app;
+ 
+await main();
+ 
+async function main()
+{
+  return new Promise((resolve, reject) => { mySnippet(); })
+}
+ 
+function mySnippet()
+{
+    //![Add event listener.]
+    var myEventListener = app.addEventListener("afterNew", myDisplayEventType);
+    //![Add event listener.]
+}
+ 
+//![Add event listener - functions.]
+function myDisplayEventType(myEvent)
+{
+    alert("This event is the " + myEvent.eventType + " event.");
+}
+ 
+function alert(msg)
+{
+    theDialog = app.dialogs.add();
+    col = theDialog.dialogColumns.add();
+    colText = col.staticTexts.add();
+    colText.staticLabel = "" + msg;
+    theDialog.canCancel = false;
+    theDialog.show();
+    theDialog.destroy();
+    return;
+}
+```
+Here is a sample events script to run and listen for "afterNew" event only once and remove, which generates an alert if a new document is created.
+
+```js
+let myInDesign = require("indesign");
+let app = myInDesign.app;
+let myResolve;
+ 
+await main();
+ 
+async function main()
+{
+  return new Promise((resolve, reject) => { mySnippet(); myResolve = resolve;});
+}
+ 
+function mySnippet()
+{
+    //![Add event listener.]
+    var myEventListener = app.addEventListener("afterNew", myDisplayEventType);
+    //![Add event listener.]
+}
+ 
+//![Add event listener - functions.]
+function myDisplayEventType(myEvent)
+{
+    alert("This event is the " + myEvent.eventType + " event.");
+ 
+    //![remove event listener.]
+    app.removeEventListener("afterNew", myDisplayEventType);
+    //![remove event listener.]
+ 
+    myResolve();
+}
+ 
+function alert(msg)
+{
+    theDialog = app.dialogs.add();
+    col = theDialog.dialogColumns.add();
+    colText = col.staticTexts.add();
+    colText.staticLabel = "" + msg;
+    theDialog.canCancel = false;
+    theDialog.show();
+    theDialog.destroy();
+    return;
+}
+```
+
+## Menus
+InDesign scripting can add or remove menu items, perform any menu command, and attach scripts to menu items. This section shows how to work with InDesign menu scripting.
```

Comment:
Avoid reference to 'scripting' if we know this will work in plugins as well.

Reply:
Agreed